src/ssl.c

```diff
@@ -4151,32 +4151,41 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
 #if defined(OPENSSL_EXTRA) || defined(HAVE_WEBSERVER)
     {
         /* remove encrypted header if there */
-        char encHeader[] = "Proc-Type";
-        char* line = XSTRNSTR(headerEnd, encHeader, PEM_LINE_LEN);
-        if (line) {
-            char* newline;
-            char* finish;
-            char* start  = XSTRNSTR(line, "DES", PEM_LINE_LEN);
-
-            if (!start)
-                start = XSTRNSTR(line, "AES", PEM_LINE_LEN);
-
-            if (!start) return SSL_BAD_FILE;
-            if (!info)  return SSL_BAD_FILE;
-
-            finish = XSTRNSTR(start, ",", PEM_LINE_LEN);
-
-            if (start && finish && (start < finish)) {
-                newline = XSTRNSTR(finish, "\r", PEM_LINE_LEN);
+        char   encHeader[] = "Proc-Type";
+        word32 headerEndSz = (word32)(bufferEnd - headerEnd);
+        char*  line        = XSTRNSTR(headerEnd, encHeader, min(headerEndSz,
+                                                                PEM_LINE_LEN));
+        if (line != NULL) {
+            char*  newline;
+            char*  finish;
+            word32 finishSz;
+            word32 startSz;
+            word32 lineSz   = (word32)(bufferEnd - line);
```

Comment:
`line` comes as the output of `XSTRNSTR(headerEnd, encHeader, min(headerEndSz, PEM_LINE_LEN))`. By the nature of `XSTRNSTR()`, `line` is guaranteed to either be `NULL`, or it will be a pointer greater than or equal to `headerEnd` and strictly less than `headerEnd + headerEndSz`. Note that this is true regardless of the value of `PEM_LINE_LEN`.

`headerEndSz` comes from the expression `bufferEnd - headerEnd`, meaning that `headerEnd + headerEndSz` will be the same as `bufferEnd` and `line` will be strictly less than `bufferEnd` if it is not `NULL`

This is, of course, assuming that `bufferEnd` and `headerEnd` are no more than `UINT32_MAX` characters apart, but I didn't make this assumption first: `sz`, the effective size of the buffer, is an `int`, meaning the buffer itself is assumed not to be longer that `INT_MAX` characters long. Because `bufferEnd` is set as `buff + sz`, this more restrictive assumption carries down to my code.

`headerEnd` comes from `XSTRNSTR((char*)buff, header, sz)`, and again by nature of `XSTRNSTR()` it will either be `NULL`, or it will be greater than or equal to `buff` and strictly less than `buff + sz`. Recalling that `bufferEnd` is defined as `buff + sz`, `headerEnd` is guaranteed to be strictly less than `bufferEnd` if it is not `NULL`.

I am careful to never use a `NULL` pointer when calculating the sizes, meaning that when `lineSz` is calculated, `line` is guaranteed to be strictly less than `bufferEnd`, and similar confidence extends to the other two instances you pointed out.

Regardless of the above, it all depends on `XSTRNSTR()` to be well behaved. By the nature of it, `XSTRNSTR()` can exchanged for anyone's implementation when wolfSSL is compiled. If that implementation is poorly behaved, then the guarantee I relied upon above may be broken; though I would claim this would be a bug in `XSTRNSTR()`'s implementation rather than our library.

With all that in mind, I claim that `line` is guaranteed to be strictly less than `bufferEnd` and there isn't any error, but I will defer to your judgment. Do you still want me to explicitly check if `line` is less than `bufferEnd` before calculating `lineSz`?